### DISCUSSION TITLE
veb , x.vweb , vweb Which one should I use?

### DISCUSSION THREAD

**Discussion by Jengro777** [vlang/v#21550] [Questions and Answers] [answered]
`veb: copy x.vweb to veb, and make it work with comptime`

1、What is veb?  Will it replace x.vweb?
2、Will vweb be replaced by veb later?  I want to determine which web library I should use in the future

**Comment by JalonSolov** [comment]
veb is the latest.  vweb is the oldest, and is deprecated (whether you see a deprecation message or not).

x.vweb is where veb came from, and it is also deprecated (but both vweb and x.vweb will stay for a while so older code that uses them won't immediately break).

**Reply by Jengro777** [reply]
ok

**Comment by l-33ter** [comment]
Then, it would be useful when there is a 'deprecation' notification on github? - If you say "Yes"! I will do it!

**Reply by JalonSolov** [reply, marked as answer]
`vlib/vweb` and `vlib/x.vweb` are subdirs of the V repo.

V itself should be giving deprecation messages when they are imported, but module-level deprecation messages aren't working right now.

**Comment by exvillager** [comment]
veb is latest and officail way. all those two are older and deprecated. use veb
